src/WebJobs.Script/Workers/Rpc/Configuration/LanguageWorkerOptionsSetup.cs

```diff
@@ -18,18 +20,21 @@ internal class LanguageWorkerOptionsSetup : IConfigureOptions<LanguageWorkerOpti
         private readonly IEnvironment _environment;
         private readonly IMetricsLogger _metricsLogger;
         private readonly IWorkerProfileManager _workerProfileManager;
+        private readonly IScriptHostManager _scriptHostManager;
 
         public LanguageWorkerOptionsSetup(IConfiguration configuration,
                                           ILoggerFactory loggerFactory,
                                           IEnvironment environment,
                                           IMetricsLogger metricsLogger,
-                                          IWorkerProfileManager workerProfileManager)
+                                          IWorkerProfileManager workerProfileManager,
+                                          IScriptHostManager scriptHostManager)
```

Comment:
can't this run at JobHost-only level too (without a WebHost)? Is there a ScriptHostManager in that scenario?

Reply:
Yes, When running on WebHost level, there is a ScriptHostManager instance, but the ActiveHost property is null.